wandb/apis/file_stream.py

```diff
@@ -29,10 +31,36 @@ def process_chunks(self, chunks):
         }
 
 
-class OverwriteFilePolicy(object):
+class HistoryFilePolicy(object):
+    def __init__(self, start_chunk_id=0):
+        self._chunk_id = start_chunk_id
+
     def process_chunks(self, chunks):
+        chunk_id = self._chunk_id
+        self._chunk_id += len(chunks)
+        chunk_data = []
+        for chunk in chunks:
+            if len(chunk.data) + 1 > MAX_SUMMARY_HISTORY_LINE_SIZE:
+                wandb.termerror('History line is {} bytes but maximum size is {} bytes. Truncating to the maximum size. Some data may be missing from plots.'.format(len(chunk.data), MAX_SUMMARY_HISTORY_LINE_SIZE - 1))
+                chunk_data.append(chunk.data[:MAX_SUMMARY_HISTORY_LINE_SIZE - 1])
```

Comment:
We can't truncate here because it will send invalid JSON to the backend.

Reply:
Hmm do you have a better suggestion? I figured this is best effort. The backend currently doesn’t crash on invalid JSON. I dunno if I really want to explore the object looking for which part is too big but maybe that would be a way to go

Reply:
Thought of another idea: truncate the JSON, parse it using a streaming parser, and put what it finds into a new JSON object

Reply:
That could work...  is there a built in streaming parser?  We could also just drop the whole row and warn the user.

Reply:
Yeah I think there’s a pretty standard streaming parser I could use